x/group/keeper/invariants_test.go

```diff
@@ -0,0 +1,594 @@
+package keeper_test
+
+import (
+	"testing"
+	"time"
+
+	gogotypes "github.com/gogo/protobuf/types"
+	"github.com/stretchr/testify/suite"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	dbm "github.com/tendermint/tm-db"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/store"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/group"
+	"github.com/cosmos/cosmos-sdk/x/group/internal/orm"
+	"github.com/cosmos/cosmos-sdk/x/group/keeper"
+)
+
+type invariantTestSuite struct {
+	suite.Suite
+}
+
+func TestInvariantTestSuite(t *testing.T) {
+	suite.Run(t, new(invariantTestSuite))
+}
+
+func (s *invariantTestSuite) SetupSuite() {
+	s.T().Parallel()
+}
+
+func getCtxCodecKey() (sdk.Context, *codec.ProtoCodec, *storetypes.KVStoreKey) {
+	interfaceRegistry := types.NewInterfaceRegistry()
+	group.RegisterInterfaces(interfaceRegistry)
+	cdc := codec.NewProtoCodec(interfaceRegistry)
+	key := sdk.NewKVStoreKey(group.ModuleName)
+	db := dbm.NewMemDB()
+	cms := store.NewCommitMultiStore(db)
+	cms.MountStoreWithDB(key, storetypes.StoreTypeIAVL, db)
+	_ = cms.LoadLatestVersion()
+	// s.Require().NoError(err)
+	curCtx := sdk.NewContext(cms, tmproto.Header{}, false, log.NewNopLogger())
+	curCtx = curCtx.WithBlockHeight(10)
+	return curCtx, cdc, key
+}
+
+func (s *invariantTestSuite) TestTallyVotesInvariant() {
+	curCtx, cdc, key := getCtxCodecKey()
+	prevCtx, _ := curCtx.CacheContext()
+	prevCtx = prevCtx.WithBlockHeight(curCtx.BlockHeight() - 1)
+
+	// Proposal Table
+	proposalTable, err := orm.NewAutoUInt64Table([2]byte{keeper.ProposalTablePrefix}, keeper.ProposalTableSeqPrefix, &group.Proposal{}, cdc)
```

Comment:
since the group keeper is public (compared to the private `serverImpl` we had in regen-ledger), you could just use the existing testsuite in keeper_test.go and run the tests on the main invariants function (that take `keeper` as param)
that would avoid having to setup the ctx, cdc, db and the table in those invariants tests
you can always take the initial context and cache it so these invariants tests don't interfere with the other tests in keeper_test.go